lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,13 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        if e.message =~ /8007089A/m
```

Comment:
Ah I see why you did this now.  I think we got our lines crossed... let me think about this.

Perhaps we do only want to catch `WIN32OLERuntimeError` and not *all* exceptions.